src/pages/Sketch/components/Canvas.jsx

```diff
@@ -493,27 +498,45 @@ function Canvas() {
       });
   };
 
-  const initiateLoadSaved = () => document.getElementById("file-selector").click();
-  const loadLastState = e => {
+
+  
+ 
+  const initiateload = () => document.getElementById("file-selector").click();
+  const loadimport= e =>{   
     let file = e.target.files[0];
-    if (!file) return;
-    let reader = new FileReader();
+    var textFile = /image.*/;
+
+
+    if (!(file.type.match(textFile))) { 
+     alert("OOPs! You didn't select an image. We only support jpg,png and webp");
+        file.value = '';
+        return false;
+            }
+            else
+            {
+    let reader =  new FileReader();
```

Comment:
d2ps is loaded as an image

d2ps is not an image
Its a raw buffer

code is modified blindly and thus d2ps feature broke


your should have CREATED YOUR OWN FUNCTION instead of MODIFYING THE CURRENT FUNCTION BLINDLY